### PR TITLE
refactor: migrate submissions metadata

### LINKED ISSUE #1520: [API] Migrate GET submission metadata related endpoints to /api counterpart
This issue tracks the migration of `GET /:formId/adminform/submissions/` and `GET /:formId/adminform/submissions/metadata` to their new `/api` counterparts.

This issue is distinct from #1511 due to the request query parameter shifting into the endpoint itself, which requires a new controller to be written instead of just duplicating the route. 
This change is proposed as the current endpoints are not RESTful, as queries should be for properties of a resource, not the identifier for a resource.

\# | Existing endpoints | New endpoints<br>(note that all routes are prefixed with /api/v3) | Description
-- | -- | --|--|
1 | GET /:formId/adminform/submissions/<br>{ query: { submissionId } } | GET /admin/forms/:formId/submissions/:id | Retrieve submission by id
2 | GET /:formId/adminform/submissions/metadata<br>{ query: { submissionId, page } } | GET /admin/forms/:formId/submissions/metadata<br>{ query: { searchId?, page = 1 } } | Search by submission

> For the first endpoint, note the inlining of `query.submissionId` to the resource identifier `:id` in the endpoint itself.

The second endpoint just has a change in query params from `submissionId` to `searchId` for better clarity as to what the query is used for

### PR DESCRIPTION
**Pull request by seaerchin**
This is part 2 of #1520. 

## Problem
This PR migrates the old API to the new API rooted at `/api/v3/admin/forms`. The current implementation is ported over as is and something to note is that currently, we are not validating that there is either submission or formId, and that formId should be at least 1. 

This will be rectified in a future PR to avoid scope creep here and this PR should be reviewed as an almost direct port over. 

Closes #1520 

## Solution
1. Adds new endpoint to `AdminFormSubmissions` router for metadata and changed client callsites
2. Ported over integration tests

## Manual Tests

- [ ] Create a storage mode form and activate it. Submit **>10** responses to the form. Go to the admin panel and click on data
    - Verify that clicking on any page returns a list of responses  
    - Verify that searching on any single response id returns exactly that response   
    - Verify that searching on any substring of a response id returns nothing

### CODE REVIEW

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts
Comment (mantariksh) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:721]: can we restrict the scope of this PR to the API refactor only? let's not repeat the mistakes of the previous production bug by mixing refactors and logic changes

**Review by mantariksh — COMMENTED** [review]
tested on staging?

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts
Comment (seaerchin) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:721]: sure, i'll revert this change to adhere to before

**Comment by seaerchin** [comment]
@mantariksh tested on local, haven't tested on staging. will borrow staging tomorrow and test

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts
Comment (karrui) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:686]: nit: if this is only used in one place, can we inline it instead?

**Review by karrui — APPROVED** [review]
lgtm, can squash after fixing PR according to anti's comments

**Comment by mantariksh** [comment]
pls test on staging before merging

**Comment by seaerchin** [comment]
I got 429 `too many requests` while testing. this occurred at the start of testing, when clicking on teh second page of data. Didn't occur later on after refreshing the page and retrying. 

I don't think the error is due to this PR and more likely due to middleware set up for rate limiting.

**Comment by seaerchin** [comment]
tested on staging again; no issues found. will merge into develop

**Pull request merged**
